### ISSUE TITLE
User/Group setting ignored

### ISSUE THREAD

**Issue by bastelfreak** [elastic/puppet-elasticsearch#295]
We use the following profile:
```
class profiles::elasticsearch {
  class { '::elasticsearch':
    repo_version  => '1.4',
    java_install  => true,
    manage_repo   => true,
    config        => { 'cluster.name' => 'elasticsearch_cluster_1' },
    elasticsearch_user  => 'elasticsearch',
    elasticsearch_group => 'elasticsearch',
  }
  elasticsearch::instance { 'es-01': 
    config  => { 'node.name' => $::hostname },
    init_defaults => {
                        'ES_USER'   => 'elasticsearch', 
                        'ES_GROUP'  => 'elasticsearch', 
                        'user'      => 'elasticsearch', 
                        'group'     => 'elasticsearch'},
  }
  elasticsearch::plugin { 'lmenezes/elasticsearch-kopf':
    module_dir  => 'kopf',
    instances   => 'es-01',
    proxy_host  => 'proxy...de',
    proxy_port  => 3128,
  }
}
```

unfortunatly elasticsearch ignores the user/group setting for the instance and starts it as root. we are using it on centos7

**Comment by bastelfreak** [comment]
to clarify the issue, the "User= " and "Group= " setting in the systemd service file is empty

**Comment by electrical** [comment]
Could you try with https://github.com/elastic/puppet-elasticsearch/pull/296 applied to see if it fixes the issue?

**Comment by bastelfreak** [comment]
works perfectly:
```
Notice: /Stage[main]/Profiles::Elasticsearch/Elasticsearch::Instance[es-01]/Elasticsearch::Service[es-01]/Elasticsearch::Service::Systemd[es-01]/File[/usr/lib/systemd/system/elasticsearch-es-01.service]/content: 
--- /usr/lib/systemd/system/elasticsearch-es-01.service	2015-03-18 11:45:14.662054495 +0100
+++ /tmp/puppet-file20150324-29706-176gqgj	2015-03-24 09:04:40.669241656 +0100
@@ -5,8 +5,8 @@
 [Service]
 Type=forking
 EnvironmentFile=/etc/sysconfig/elasticsearch-es-01
-User=
-Group=
+User=elasticsearch
+Group=elasticsearch
 PIDFile=/var/run/elasticsearch/elasticsearch-es-01.pid
 ExecStart=/usr/share/elasticsearch/bin/elasticsearch -d -p /var/run/elasticsearch/elasticsearch-es-01.pid -Des.default.config=$CONF_FILE -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.work=$WORK_DIR -Des.default.path.conf=$CONF_DIR
 # See MAX_OPEN_FILES in sysconfig
```

**Comment by bastelfreak** [comment]
This leads to another error, elasticsearch isn't able to start:

```
# journalctl -u elasticsearch-es-01.service --no-pager --full -n 10
-- Logs begin at Thu 2015-03-12 12:03:15 CET, end at Tue 2015-03-24 09:10:57 CET. --
Mar 24 09:08:27 es1.vps.hosteurope.de systemd[1]: Starting Starts and stops a single elasticsearch instance on this system...
Mar 24 09:08:27 es1.vps.hosteurope.de systemd[1]: PID file /var/run/elasticsearch/elasticsearch-es-01.pid not readable (yet?) after start.
Mar 24 09:08:27 es1.vps.hosteurope.de systemd[1]: Started Starts and stops a single elasticsearch instance on this system.
Mar 24 09:08:28 es1.vps.hosteurope.de systemd[1]: elasticsearch-es-01.service: main process exited, code=exited, status=3/NOTIMPLEMENTED
Mar 24 09:08:28 es1.vps.hosteurope.de systemd[1]: Unit elasticsearch-es-01.service entered failed state.
Mar 24 09:10:56 es1.vps.hosteurope.de systemd[1]: Starting Starts and stops a single elasticsearch instance on this system...
Mar 24 09:10:56 es1.vps.hosteurope.de systemd[1]: PID file /var/run/elasticsearch/elasticsearch-es-01.pid not readable (yet?) after start.
Mar 24 09:10:57 es1.vps.hosteurope.de systemd[1]: Started Starts and stops a single elasticsearch instance on this system.
Mar 24 09:10:57 es1.vps.hosteurope.de systemd[1]: elasticsearch-es-01.service: main process exited, code=exited, status=3/NOTIMPLEMENTED
Mar 24 09:10:57 es1.vps.hosteurope.de systemd[1]: Unit elasticsearch-es-01.service entered failed state.
```
```
systemctl status elasticsearch-es-01.service 
elasticsearch-es-01.service - Starts and stops a single elasticsearch instance on this system
   Loaded: loaded (/usr/lib/systemd/system/elasticsearch-es-01.service; enabled)
   Active: failed (Result: exit-code) since Tue 2015-03-24 09:10:57 CET; 3min 4s ago
     Docs: http://www.elasticsearch.org
  Process: 30436 ExecStart=/usr/share/elasticsearch/bin/elasticsearch -d -p /var/run/elasticsearch/elasticsearch-es-01.pid -Des.default.config=$CONF_FILE -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.work=$WORK_DIR -Des.default.path.conf=$CONF_DIR (code=exited, status=0/SUCCESS)
 Main PID: 30442 (code=exited, status=3)

Mar 24 09:10:56 es1.vps.hosteurope.de systemd[1]: PID file /var/run/elasticsearch/elasticsearch-es-01.pid not readable (yet?) after start.
Mar 24 09:10:57 es1.vps.hosteurope.de systemd[1]: Started Starts and stops a single elasticsearch instance on this system.
Mar 24 09:10:57 es1.vps.hosteurope.de systemd[1]: elasticsearch-es-01.service: main process exited, code=exited, status=3/NOTIMPLEMENTED
Mar 24 09:10:57 es1.vps.hosteurope.de systemd[1]: Unit elasticsearch-es-01.service entered failed state.
```

```
[Unit]
Description=Starts and stops a single elasticsearch instance on this system
Documentation=http://www.elasticsearch.org

[Service]
Type=forking
EnvironmentFile=/etc/sysconfig/elasticsearch-es-01
User=elasticsearch
Group=elasticsearch
PIDFile=/var/run/elasticsearch/elasticsearch-es-01.pid
ExecStart=/usr/share/elasticsearch/bin/elasticsearch -d -p /var/run/elasticsearch/elasticsearch-es-01.pid -Des.default.config=$CONF_FILE -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.work=$WORK_DIR -Des.default.path.conf=$CONF_DIR
# See MAX_OPEN_FILES in sysconfig
LimitNOFILE=65535
# See MAX_LOCKED_MEMORY in sysconfig, use "infinity" when MAX_LOCKED_MEMORY=unlimited and using bootstrap.mlockall: true
#LimitMEMLOCK=infinity
# Shutdown delay in seconds, before process is tried to be killed with KILL (if configured)
TimeoutStopSec=20

[Install]
WantedBy=multi-user.target
```

**Comment by bastelfreak** [comment]
First issue: elasticsearch can't write to its logfile
```chown elasticsearch:elasticsearch /var/log/elasticsearch/es-01/elasticsearch_cluster_1.log```

and it can't access his data:
```chown elasticsearch:elasticsearch /usr/share/elasticsearch/data/es-01/elasticsearch_cluster_1 -R```

After that, everthing works fine:
```
systemctl start elasticsearch-es-01.service ; systemctl status elasticsearch-es-01.service 
elasticsearch-es-01.service - Starts and stops a single elasticsearch instance on this system
   Loaded: loaded (/usr/lib/systemd/system/elasticsearch-es-01.service; enabled)
   Active: active (running) since Tue 2015-03-24 09:21:34 CET; 1min 26s ago
     Docs: http://www.elasticsearch.org
  Process: 30710 ExecStart=/usr/share/elasticsearch/bin/elasticsearch -d -p /var/run/elasticsearch/elasticsearch-es-01.pid -Des.default.config=$CONF_FILE -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.work=$WORK_DIR -Des.default.path.conf=$CONF_DIR (code=exited, status=0/SUCCESS)
 Main PID: 30716 (java)
   CGroup: /system.slice/elasticsearch-es-01.service
           └─30716 java -Xms256m -Xmx1g -Xss256k -Djava.awt.headless=true -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=75 -XX:+UseCMSInitiatingOccupancyOnly -XX:+HeapDumpOnOutOfM...

Mar 24 09:21:34 es1.vps.hosteurope.de systemd[1]: PID file /var/run/elasticsearch/elasticsearch-es-01.pid not readable (yet?) after start.
Mar 24 09:21:34 es1.vps.hosteurope.de systemd[1]: Started Starts and stops a single elasticsearch instance on this system.
```

**Comment by electrical** [comment]
Thank you for testing. the rights issue was expected yeah since everything was owned by root.
I'll merge the PR and do a release within a few hours.
Thanks!

**Comment by electrical** [comment]
Released as 0.9.3

Again hanks for reporting it!
